packages/engine/src/datastore/table/proxy.rs

```diff
@@ -163,14 +160,14 @@ impl Debug for StateReadProxy {
 }
 
 impl StateReadProxy {
-    pub fn new<K: ReadState>(state: &K) -> Result<Self> {
+    pub fn new(state: &State) -> Result<Self> {
         Ok(StateReadProxy {
             agent_pool_proxy: state.agent_pool().read_proxy()?,
             message_pool_proxy: state.message_pool().read_proxy()?,
         })
     }
 
-    pub fn new_partial<K: ReadState>(state: &K, indices: &[usize]) -> Result<Self> {
+    pub fn new_partial(state: &State, indices: &[usize]) -> Result<Self> {
```

Comment:
`indices` could be renamed to `group_indices` if we're changing the function anyways.

Reply:
I'd do this in my follow up PR